woocommerce/woocommerce
# Fix selector in shopper test

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





I recently updated the test in scope of https://github.com/woocommerce/woocommerce/pull/52207. The test was failing with WooPayments active.

Reason: [Editor is iframed](https://github.com/WordPress/gutenberg/issues/53511#issuecomment-1672785421) if all blocks are registered with `apiVersion: 3` (this is the case with Blocks). WooPayments registers at least one block with `apiVersion: 2` which removes the iframe so the selector I wrote became incorrect.

Fix: I switched to [`getCanvas` util](https://github.com/woocommerce/woocommerce/blob/0535d366aca8bad933e0cfad572783f135077495/plugins/woocommerce/tests/e2e-pw/utils/editor.js#L40C7-L40C16) that accounts for that. Also, I simplified the collection selection.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

### Make sure the test is passing with and without WooPayments

1. Run the environment: `pnpm --filter='@woocommerce/plugin-woocommerce' env:start`
2. Run the test in question `pnpm --filter='@woocommerce/plugin-woocommerce' test:e2e shop-products-filter` (this is equivalent to what runs on CI in the repo
3. **Expected:** Test passes
4. Go to the environment > Plugins (http://localhost:8086/wp-admin/plugins.php)
5. Install and activate WooPayments (no configuration is required)
6. Run the test again `pnpm --filter='@woocommerce/plugin-woocommerce' test:e2e shop-products-filter` (this is equivalent to what runs on CI in the repo
7. **Expected:** Test passes



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
